session-replay/Tests/DatadogSessionReplayTests/Writer/WriterTests.swift

```diff
@@ -0,0 +1,24 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2019-2020 Datadog, Inc.
+ */
+
+import XCTest
+@testable import DatadogSessionReplay
+
+// swiftlint:disable empty_xctest_method
+class WriterTests: XCTestCase {
+    func testWhenFeatureScopeIsConnected_itWritesRecordsToCore() {
+        // TODO: RUMM-2690
+        // Implementing this test requires creating mocks for `DatadogContext` (passed in `FeatureScope`),
+        // which is yet not possible as we lack separate, shared module to facilitate tests.
+    }
```

Comment:
What I mean here is that `Writer` depends on V2 `FeatureScope` which exposes the _event write context_:
```swift
func eventWriteContext(bypassConsent: Bool, _ block: @escaping (DatadogContext, Writer) throws -> Void)
```
In order to create `FeatureScopeMock: FeatureScope` that records what is being written, we have to mock `DatadogContext` (e.g. `.mockAny()`). Mocking it isn't yet straightforward as `DatadogContext` depends on several structures internal to `Datadog`. I don't want to make it all public as part of this PR - I believe it's best to be done through separate module that facilitates tests.

Reply:
Agreed! That will be the role of the `TestUtilities` module, but it's not yet available.